src/fireedge/src/modules/components/Forms/VmTemplate/CreateForm/Steps/ExtraConfiguration/context/userInputsAutocompleteDialog.js

```diff
@@ -0,0 +1,210 @@
+ * ------------------------------------------------------------------------- */
+import PropTypes from 'prop-types'
+import { Component, useState } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import {
+  VrTemplateAPI,
+  addUserInputSuggestionVR,
+  removeUserInputSuggestionVR,
+  setUserInputSuggestionsVR,
+  useGeneralApi,
+} from '@FeaturesModule'
+import {
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogActions,
+  Button,
+  TextField,
+  Card,
+  CardContent,
+  CardActions,
+  IconButton,
+  Typography,
+  Box,
+  CircularProgress,
+} from '@mui/material'
+import { Trash as DeleteIcon, Download } from 'iconoir-react'
+import { T } from '@ConstantsModule'
+
+/**
+ * @param {object} root0 - Props
+ * @param {boolean} root0.open - Is dialog open?
+ * @param {Function} root0.handleClose - Handle closing
+ * @returns {Component} - User inputs dialog
+ */
+const PopUpDialog = ({ open, handleClose }) => {
+  const [text, setText] = useState('')
+  const [value, setValue] = useState('')
+  const [error, setError] = useState(false)
+  const [fetchVRTemplates, { isFetching }] =
+    VrTemplateAPI.useLazyGetVrTemplatesQuery(undefined, {
+      skip: true,
+    })
+
+  const { enqueueError } = useGeneralApi()
+  const items = useSelector((state) => state.persistent.userInputSuggestionsVR)
+  const dispatch = useDispatch()
+
+  const validateValue = (val) => /^[a-zA-Z0-9_]+$/.test(val)
+
+  const handleSubmit = () => {
+    if (text && value && validateValue(value)) {
+      dispatch(addUserInputSuggestionVR({ text, value }))
+      setText('')
+      setValue('')
+      setError(false)
+    } else {
+      setError(true)
+    }
+  }
+
+  const handleDelete = (itemId) => {
+    dispatch(removeUserInputSuggestionVR(itemId))
+  }
+
+  const fetchSuggestions = async () => {
+    await fetchVRTemplates()
+      .unwrap()
+      .then((data) => {
+        const serviceVRouter = [data]
+          ?.flat()
+          ?.find((template) => template.NAME === 'Service Virtual Router')
+        if (serviceVRouter) {
+          const userInputs = serviceVRouter.TEMPLATE.USER_INPUTS
+          const formattedSuggestions = Object.entries(userInputs).map(
+            ([key, val]) => {
+              // eslint-disable-next-line no-unused-vars
+              const [_, type, description] = val.split('|')
+
+              return { text: `${description} (${type.trim()})`, value: key }
+            }
+          )
+          dispatch(setUserInputSuggestionsVR(formattedSuggestions))
+        }
+      })
+      .catch(() => {
+        enqueueError(T.ErrorUserInputAutocompleteFetch)
+      })
+  }
+
+  return (
+    <Dialog open={open} onClose={handleClose}>
+      <DialogTitle>Add autocomplete suggestion</DialogTitle>
+      <DialogContent>
+        <Box
+          sx={{
+            display: 'flex',
+            justifyContent: 'flex-end',
+            alignItems: 'end',
+            mb: 2,
+          }}
+        >
+          <Button
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

JSX element content is not internationalized. Application should support multiple languages through internationalization. Instead, utilize packages like &#39;i18next&#39; to internationalize your items.